pkg/guardian/handler.go

```diff
@@ -35,29 +49,29 @@ func CondChain(cf ...CondRequestBlockerFunc) RequestBlockerFunc {
 	return func(c context.Context, r Request) (bool, uint32, error) {
 		minRemaining := uint32(math.MaxUint32)
 		for _, f := range cf {
-			stop, blocked, remaining, err := f(c, r)
-			if err != nil && stop {
-				return blocked, 0, err
+			resp, remaining, err := f(c, r)
+			if err != nil {
+				// We should never continue processing or block if there is an error, to ensure this fails open
+				return false, remaining, err
```

Comment:
This is technically a change in behavior. Before, we were respecting the blocked value from the function `f`. However, this ignores that value and allows the request through regardless.

This allows this function to fail open even in the case that the whitelister fails. 

Not necessary for this PR though, so i can put this in a follow up PR if we would prefer.